src/svxlink/contrib/module_site_status/ModuleSiteStatus.tcl

```diff
@@ -0,0 +1,291 @@
+					| (($ANALOG_NEW_STATE-$ANALOG_CURRENT_STATE)>$ANALOG_HYSTERISIS)} {
+						#update the current state for next time the value is tested
+						set ANALOG_CURRENT_STATE_$i $ANALOG_NEW_STATE
+						printInfo "ANALOG Sensor $i has changed state"
+						# determine the type of sensor to figure out what to announce
+						variable CFG_ANALOG_TYPE_$i
+						set TYPE "CFG_ANALOG_TYPE_$i"
+						set TYPE [subst $$TYPE]
+						# Handle the event based on user configurations
+						switch $TYPE {
+							TEMPERATURE {
+								TEMPERATURE $i $ANALOG_NEW_STATE
+							}
+							default {
+								printInfo "ANALOG SENSOR $i is of unknown type -$TYPE"
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	
+	
+	#basic function to announce the door sensor changing status
+	proc DOORSENSOR_ANNOUNCE {sensor value} {
+		if {$value == 1} {
+			playMsg "site_door_open"
+			printInfo "Door Sensor Number $sensor indicates the door is now open"
+		} else {
+			printInfo "Door Sensor Number $sensor indicates the door is now closed"
+		}
+	}
+	
+	#basic function to announce the door sensor changing status
+	proc FUELSENSOR_ANNOUNCE {sensor value} {
+		if {$value == 1} {
+			playMsg "fuel_low"
+			printInfo "Fuel Sensor Number $sensor indicates the fuel is now low"
+		} else {
+			playMsg "fuel_filled"
+			printInfo "fuel Sensor Number $sensor indicates the fuel is now filled"
+		}
+	}
+	
+	#basic function to announce the door sensor changing status
+	proc SOLARSENSOR_ANNOUNCE {sensor value} {
+		if {$value == 1} {
+			playMsg "solar_charging"
+			printInfo "Solar charger Number $sensor indicates the system is charging"
+		} else {
+			playMsg "solar_discharging"
+			printInfo "Solar charger Number $sensor indicates the system is discharging"
+		}
+	}
+	
+	#basic function to announce the temp sensor changing status
+	proc TEMPERATURE {sensor value} {
+		# do calculations here to convert the voltage to normal units
+		# as they are read in as raw 5V/1024 values
+		# from https://learn.adafruit.com/tmp36-temperature-sensor/using-a-temp-sensor
+		# Voltage at pin in milliVolts = (reading from ADC) * (5000/1024) 
+		# Centigrade temperature = [(analog voltage in mV) - 500] / 10
+		set voltage [expr $value * 0.004882814]
+		set temperature [expr ($voltage-0.5)*100]
+		printInfo "TEMPERATURE sensor $sensor reading is $temperature degrees Celcius"
+	}
+	
+	#basic function to announce the temp sensor changing status
+	proc BATTERY_VOLTAGE {sensor value} {
+		# do calculations here to convert the voltage to normal units
+		# as they are read in as raw 5V/1024 values
+		set value [eval (5/1024)*$value]
+		printInfo "BATTERY VOLTAGE $sensor reading is $value"
+	}
+
+	# Executed when this module is being deactivated.
+	#
+	proc deactivateCleanup {} {
+		printInfo "Module deactivated"
+	}
+	
+	# check for new events
+	proc check_for_alerts {} {
+		main_every_second
+	}
+	
+	append func $module_name "::check_for_alerts";
```

Comment:
You may call main_every_second directly here. No need to jump via check_for_alters.